File: hello-kafka-unit-test/java/src/main/java/hellokafka/Producer.java
package hellokafka;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;


@Service
public class Producer {

    @Autowired
    public final KafkaTemplate<String, String> kafka = null;

    void send(String message) {
        System.out.println("PUBLISH message: $message");
        this.kafka.send("hello-topic-ut", message);
    }
}

File: hello-kafka-unit-test/kotlin/src/main/kotlin/hellokafka/Producer.kt
package hellokafka

import org.springframework.kafka.core.KafkaTemplate
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service
import java.time.LocalDateTime

@Service
class Producer(val kafka: KafkaTemplate<String, String>) {

    fun send(message: String) {
        println("PUBLISH message: $message")
        this.kafka.send("hello-topic-ut", message)
    }
}
